scripts/subsetter.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2022 National Center for Atmospheric Research and National Oceanic and Atmospheric Administration
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+import sys
+import argparse
+import logging
+import yaml
+from glob import glob
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--control', type=str,
+    default='control.yaml',
+    help='yaml control file')
+parser.add_argument('--logfile', type=str,
+    default=sys.stdout,
+    help='log file (default stdout)')
+parser.add_argument('--outdir', type=str,
+    default=None,
+    help='output directory')
+parser.add_argument('--debug', action='store_true',
+    help='set logging level to debug')
+args = parser.parse_args()
+
+"""
+Setup logging
+"""
+logging_level = logging.DEBUG if args.debug else logging.INFO
+logging.basicConfig(stream=args.logfile, level=logging_level)
+
+"""
+Read YAML control
+"""
+with open(args.control, 'r') as f:
+    control = yaml.safe_load(f)
+
+logging.debug(control)
+
+for model in control['model']:
+    logging.info('processing:' + model)
+
+    var_str = '-v '
+
+    if 'required_vars' in control['model'][model]:
+        for var in control['model'][model]['required_vars']:
+            var_str += var + ','
+
+    """
+    for dataset in control['model'][model]['mapping']:
+        for var in control['model'][model]['mapping'][dataset]:
+            var_str += var + ','
+    """
+    for var in control['model'][model]['variables']:
+        var_str += var + ','
+
+    var_str = var_str.strip(',')
+    logging.info(var_str)
+
+    file_str = os.path.expandvars(control['model'][model]['files'])
+
+    files = sorted(glob(file_str))
+
+    for file_in in files:
+        file_subdirs = file_in.split('/')
+        file_subdirs[-1] = 'subset_' + file_subdirs[-1]
+        if args.outdir is not None:
+            file_out = os.path.join(args.outdir, file_subdirs[-1])
+        else:
+            file_out = '/'.join(file_subdirs)
+        command = 'ncks -O ' + var_str + ' ' + file_in + ' ' + file_out
```

Comment:
Additionally, I think it would be good to let the user set the the deflation level argument (-L) for ncks. The deflation level could be taken in as an argument with the argument parser. This would avoid situations where a user removes a few variables and ends up with a much larger file than before because the result is not compressed.